### PR TITLE
New PBR textures

### PR DESCRIPTION
**Pull request by michou214**
**Description**
Add new PBR textures that will replace some definition in PROTOs in a future PR. 

**Suggestions**
1. By grouping several textures under the same proto, you only keep the preview of the default one . It would be better to create subfolders to classify the textures and make them all unique. All of them would be have a preview in the menu and for the user it will be easier to find the texture he needs.

2. In the [Appearances](https://cyberbotics.com/doc/guide/appearances) documentation page, the table of content should not create a link to `NAME Field Summary`. It makes the list longer and all appearances are short so it's easy to have acces to the `Field Summary`.

**Tasks**
- [x] Generate the documentation
- [x] List all new textures
- [x] List all changements
- [x] Review my commit

**PBR textures added**
- `Marble` used in `WorkTop` for kitchen.
- `Parquetry` merged all different parquetry
- `Rubber` add a second type.
- `SolarCell` used for spatial vehicle such as the `Sojourner`.
- `Pavement` merged all different pavement
- `ThreadMetalPlate` add a second type.

**Documentation**
Changes can be seen in [doc/guide/appearances](https://cyberbotics.com/doc/guide/appearances?version=michou214:feature-new-PBR-textures)

### CODE REVIEW

File: projects/appearances/protos/Pavement.proto [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#VRML_SIM R2020a utf8
+# license: Apache License 2.0
+# license url: http://www.apache.org/licenses/LICENSE-2.0
+# A pavement material. The type can be choose between "old_small", "old_big", "standard", "slate", "regular" and "grid". The color can be overridden using the `colorOverride` field.
+
+PROTO Pavement [
+  field SFString{"old_small", "old_big", "standard", "slate", "regular", "grid"} type             "standard" # Defines the pavement type.
```
Comment (DavidMansolino) [inline projects/appearances/protos/Pavement.proto]: you should avoid `_` just use spaces (they are replaced by the `gsub` a few line later).

File: projects/appearances/protos/SolarCell.proto [hunk 2]
```diff
@@ -0,0 +1,49 @@
+#VRML_SIM R2020a utf8
+# license: Apache License 2.0
+# license url: http://www.apache.org/licenses/LICENSE-2.0
+# tags: static
```
Comment (DavidMansolino) [inline projects/appearances/protos/SolarCell.proto]:
```suggestion
```

File: projects/objects/factory/conveyors/controllers/conveyor_belt/conveyor_belt.c [hunk 3]
```diff
@@ -14,6 +14,10 @@
  * limitations under the License.
  */
 
+
+
+
+
```
Comment (DavidMansolino) [inline projects/objects/factory/conveyors/controllers/conveyor_belt/conveyor_belt.c]:
```suggestion
```

File: src/webots/nodes/WbTrack.cpp [hunk 4]
```diff
@@ -614,7 +614,7 @@ void WbTrack::prePhysicsStep(double ms) {
 
   // texture animation
   if (mTextureTransform) {
-    mTextureTransform->translate(mSurfaceVelocity * mTextureAnimationField->value());
+    mTextureTransform->translate(0.001 * ms * mSurfaceVelocity * mTextureAnimationField->value());
```
Comment (DavidMansolino) [inline src/webots/nodes/WbTrack.cpp]: This is not related and should not be included in this PR.

File: projects/samples/devices/controllers/track_conveyor_belt/track_conveyor_belt.c [hunk 5]
```diff
@@ -18,23 +18,32 @@
  * Description:   An example of a controller using the Track node to model a conveyor belt.
  */
 
-#include <stdio.h>
```
Comment (DavidMansolino) [inline projects/samples/devices/controllers/track_conveyor_belt/track_conveyor_belt.c]: This is not related and should not be included in this PR.

**Comment by michou214** [comment]
> These textures should probably be removed:
> * projects/samples/devices/worlds/textures/...

Good point, they escaped my vigilance.

> Why is there some changes in these texutres?
> * projects/samples/rendering/worlds/textures/...

The textures used are now under the `Pavement `PROTO. I renammed the texture `salamander_specular` in `salamander_rougness` to be consistent with the rest.
 
> It seems that the new textures added in #1457 have not been integrated here.

Parquet has been added in the `Parquetry `PROTO (type _light strip_) but for wood, it would be necessary to create a PROTO `Wood `with several types. This modification surely has more impact than the rest. This modification should be made in a future PR.

**Comment by DavidMansolino** [comment]
> > These textures should probably be removed:
> > 
> > * projects/samples/devices/worlds/textures/...
> 
> Good point, they escaped my vigilance.
> 
> > Why is there some changes in these texutres?
> > 
> > * projects/samples/rendering/worlds/textures/...
> 
> The textures used are now under the `Pavement `PROTO. I renammed the texture `salamander_specular` in `salamander_rougness` to be consistent with the rest.

If they are in the `Pavement` PROTO then the original files need to be removed (you should not leave unuesed texture files as they increase uselessly the Webots size).

> > It seems that the new textures added in #1457 have not been integrated here.
> 
> Parquet has been added in the `Parquetry `PROTO (type _light strip_) but for wood, it would be necessary to create a PROTO `Wood `with several types. This modification surely has more impact than the rest. This modification should be made in a future PR.

In that case the original `_light strip_` files should be removed.

File: docs/reference/changelog-r2020.md [hunk 6]
```diff
@@ -9,6 +9,13 @@ Released on XXX.
     - Added an `OfficeChair` PROTO object.
     - Added a `MetalStainlessSteelCable` appearance.
     - Added a `ConveyorPlatform` PROTO object.
+    - Added a `Marble`appearance.
+    - Added a `SolarCell` appearance.
+    - Added a `Pavement` appearance which merge deprecated `SlatePavement`, `SquarePavement` and `StonePavement` appearance.
+    - Added a second appearance in the `Rubber` appearance.
+    - Added a second appearance in the `ThreadMetalPlate` appearance.
+    - Added three appearances in `Parquetry` appearance and deprecate `ChequeredParquetry` and `DarkParquetry` appearance.
```
Comment (DavidMansolino) [inline docs/reference/changelog-r2020.md]:
```suggestion
    - Added several new appearances: `Marble`, `MetalStainlessSteelCable`, `Pavement`, `Parquetry`, and `SolarCell.
    - Added a `ConveyorPlatform` PROTO object.
```

Then in a cleanup section add:
```
  - Deprecated the following appearances: `ChequeredParquetry`, `DarkParquetry`, `SlatePavement`, `SquarePavement` and `StonePavement`.
```

And in the `Enhancements`:
```
    - Added a `type` field to the following appearances: `Rubber` and `ThreadMetalPlate`.
```

File: projects/objects/kitchen/components/protos/Worktop.proto [hunk 7]
```diff
@@ -4,56 +4,41 @@
...
-  field SFString   name          "worktop"
-  field SFVec3f    size          0.44 0.06 0.7           # Defines the size of the worktop.
-  field MFString   textureUrl    "textures/worktop.jpg"  # Defines the texture used for the worktop.
-  field SFVec2f    tileSize      0.7 0.7                 # Defines the size of texture used for the worktop.
+  field SFVec3f    translation       0 0.71 0
+  field SFRotation rotation          0 1 0 0
+  field SFString   name              "worktop"
+  field SFVec3f    size              0.4 0.06 0.7  # Defines the size of the worktop.
+  field SFFloat    mass              0             # Defines the mass of the wooden box in kg. A value smaller or equal to 0 remove the physics of the wooden box.
+  field SFFloat    IBLStrength       1             # Defines the strength of ambient lighting from the Background node.
```
Comment (DavidMansolino) [inline projects/objects/kitchen/components/protos/Worktop.proto]: Those two fields are useless, instead an `appearance` field should be opened.
And this should be mentioned in the changelog.

**Comment by DavidMansolino** [comment]
:eyes:

File: projects/appearances/protos/StonePavement.proto [hunk 8]
```diff
@@ -12,34 +13,39 @@ PROTO StonePavement [
 ]
 {
 %{
-  local textureName = fields.type.value:gsub("%s+", "_")
+  io.stderr:write("Deprecated 'StonePavement' PROTO, please use the 'Pavement' PROTO instead and select the right type.\n")
+  if fields.type.value == "recent" then
+    local textureName = "regular"
+  else
+    local textureName = "old_big"
```
Comment (DavidMansolino) [inline projects/appearances/protos/StonePavement.proto]: This is not working because the variable is local to the if and therefore you can't access it later.
This is clearly visible when trying to load it:
```
ERROR: '/home/david/webots-michou/projects/appearances/protos/StonePavement.proto': error: Template engine error: luaL_dostring error : [string "-- template script to parse a procedural node..."]:51: Runtime error in sandboxed code "StonePavement.proto" in line 14:
 attempt to concatenate global 'textureName' (a nil value)

 11:        url [\
 12:          ")
 13:  __liluat_output_function( "\"textures/pavement/" .. textureName .. "_pavement_base_color.jpg\"" )
 14:> __liluat_output_function("\
 15:        ]\
 16:        filtering 5\
 17:      }\.
```

File: projects/appearances/protos/Parquetry.proto [hunk 9]
```diff
@@ -1,39 +1,56 @@
 #VRML_SIM R2020a utf8
 # license: Apache License 2.0
 # license url: http://www.apache.org/licenses/LICENSE-2.0
-# A wooden material in a tesselated block pattern covered with a layer of varnish. Useful with the `Floor` PROTO.
+# A parquetry material. The type can be choose between "chequered", "dark strip", "light strip" and "mosaic". The color can be overridden using the `colorOverride` field.
```
Comment (DavidMansolino) [inline projects/appearances/protos/Parquetry.proto]:
```suggestion
# A parquetry material.
# The type of parquetry can be set using the `type` field.
# The color can be overridden using the `colorOverride` field.
```

File: projects/appearances/protos/Parquetry.proto [hunk 10]
```diff
@@ -1,39 +1,56 @@
...
+  field SFColor                                                      colorOverride    1 1 1     # Defines the default color multiplied with the texture color.
+  field SFNode                                                       textureTransform NULL      # Defines an optional 2d texture transform.
+  field SFFloat                                                      IBLStrength      1         # Defines the strength of ambient lighting from the Background node.
 ]
 {
+%{
+  local textureName = fields.type.value:gsub("%s+", "_")
+  local extension = "jpg\""
+  if fields.type.value == "light strip" then
+    extension = "png\""
```
Comment (DavidMansolino) [inline projects/appearances/protos/Parquetry.proto]: Adding the final `"` here is extremely misleading.

File: projects/appearances/protos/Parquetry.proto [hunk 11]
```diff
@@ -1,39 +1,56 @@
...
+  end
+}%
   PBRAppearance {
     baseColor IS colorOverride
     baseColorMap ImageTexture {
       url [
-        "textures/parquetry/parquetry_base_color.jpg"
+        %{= "\"textures/parquetry/" .. textureName .. "_parquetry_base_color." .. extension }%
       ]
+      filtering 5
```
Comment (DavidMansolino) [inline projects/appearances/protos/Parquetry.proto]: Is there a reason for adding all these filtering?

File: projects/appearances/protos/Pavement.proto [hunk 12]
```diff
@@ -0,0 +1,49 @@
+#VRML_SIM R2020a utf8
+# license: Apache License 2.0
+# license url: http://www.apache.org/licenses/LICENSE-2.0
+# A pavement material. The type can be choose between "old_small", "old_big", "standard", "slate", "regular" and "grid". The color can be overridden using the `colorOverride` field.
```
Comment (DavidMansolino) [inline projects/appearances/protos/Pavement.proto]:
```suggestion
# A pavement material.
# The type of parquetry can be set using the `type` field.
# The color can be overridden using the `colorOverride` field.
```

File: projects/appearances/protos/Rubber.proto [hunk 13]
```diff
@@ -1,35 +1,36 @@
 #VRML_SIM R2020a utf8
 # license: Apache License 2.0
 # license url: http://www.apache.org/licenses/LICENSE-2.0
-# A rubber material.
+# A rubber material. The type can be choose between "flat" and "dotted".
```
Comment (DavidMansolino) [inline projects/appearances/protos/Rubber.proto]:
```suggestion
# A rubber material.
# The type of parquetry can be set using the `type` field.
```

File: projects/appearances/protos/Parquetry.proto (hunk 10 above)
Comment (DavidMansolino) [reply projects/appearances/protos/Parquetry.proto]: In any case you can probably use `jpg` for all of them.

File: projects/appearances/protos/Pavement.proto [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+PROTO Pavement [
+  field SFString{"old small", "old big", "standard", "slate", "regular", "grid"} type             "standard" # Defines the pavement type.
```
Comment (DavidMansolino) [inline projects/appearances/protos/Pavement.proto]: `regular` is not a good name, what is a 'regular' pavement? This is not really descriptive?

File: projects/appearances/protos/SolarCell.proto [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    normalMapFactor 1.5
+    IBLStrength IS IBLStrength
+    occlusionMap ImageTexture {
+      url [
+        "textures/solar_cell/solar_cell_occlusion.png"
+      ]
+      filtering 5
+    }
+    occlusionMapStrength 1
```
Comment (DavidMansolino) [inline projects/appearances/protos/SolarCell.proto]:
```suggestion
```

Useless, default value is already 1.

File: projects/appearances/protos/SolarCell.proto [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+  field SFNode  textureTransform  NULL   # Defines an optional 2d texture transform.
+  field SFFloat IBLStrength       1      # Defines the strength of ambient lighting from the Background node.
+]
+{
+  PBRAppearance {
+    baseColorMap ImageTexture {
+      url [
+        "textures/solar_cell/solar_cell_base_color.jpg"
+      ]
+      filtering 5
```
Comment (DavidMansolino) [inline projects/appearances/protos/SolarCell.proto]: Is there any reason for all these filtering?

File: projects/appearances/protos/ThreadMetalPlate.proto [hunk 17]
```diff
@@ -4,36 +4,37 @@
 # A worn paint layer on a metal surface. Useful with the `PipeSection` PROTO or any painted industrial surfaces.
```
Comment (DavidMansolino) [inline projects/appearances/protos/ThreadMetalPlate.proto]:
```suggestion
# A worn paint layer on a metal surface.
```

File: projects/objects/kitchen/components/protos/Worktop.proto [hunk 18]
```diff
@@ -1,59 +1,29 @@
...
-  field SFVec3f    translation   0 0.71 0
-  field SFRotation rotation      0 1 0 0
-  field SFString   name          "worktop"
-  field SFVec3f    size          0.44 0.06 0.7           # Defines the size of the worktop.
-  field MFString   textureUrl    "textures/worktop.jpg"  # Defines the texture used for the worktop.
-  field SFVec2f    tileSize      0.7 0.7                 # Defines the size of texture used for the worktop.
+  field SFVec3f    translation  0 0.71 0
+  field SFRotation rotation     0 1 0 0
+  field SFString   name         "worktop"
+  field SFVec3f    size         0.4 0.06 0.7  # Defines the size of the worktop.
```
Comment (DavidMansolino) [inline projects/objects/kitchen/components/protos/Worktop.proto]:
```suggestion
  field SFVec3f    size         0.44 0.06 0.7  # Defines the size of the worktop.
```

You should not change the default size.

File: projects/objects/kitchen/components/protos/Worktop.proto [hunk 19]
```diff
@@ -1,59 +1,29 @@
...
+  field SFNode     appearance   Marble{ }     # Defines the appearance.
```
Comment (DavidMansolino) [inline projects/objects/kitchen/components/protos/Worktop.proto]:
```suggestion
  field SFNode     appearance   Marble { }     # Defines the appearance.
```

File: scripts/icon_studio/controllers/icon_creator/appearances.json [hunk 20]
```diff
@@ -11,9 +11,6 @@
   "CarpetFibers": {
     "fields": "colorOverride 0.33 0.34 0.33 textureTransform TextureTransform { scale 1 0.5 }"
   },
-  "ChequeredParquetry": {
-    "fields": "textureTransform TextureTransform { scale 5 2 }"
-  },
```
Comment (DavidMansolino) [inline scripts/icon_studio/controllers/icon_creator/appearances.json]: Even if they are deprecated you should leave them here, they should be removed once actually removed.

File: projects/appearances/protos/Parquetry.proto (hunk 11 above)
Comment (michou214) [reply projects/appearances/protos/Parquetry.proto]: Some yes. I check them all again and removed useless one.

**Comment by DavidMansolino** [comment]
👀

File: projects/appearances/protos/Pavement.proto [hunk 21]
```diff
@@ -0,0 +1,48 @@
+#VRML_SIM R2020a utf8
+# license: Apache License 2.0
+# license url: http://www.apache.org/licenses/LICENSE-2.0
+# A pavement material.
+# The type of pavement can be set using the `type` field.
+# The color can be overridden using the `colorOverride` field.
+
+PROTO Pavement [
+  field SFString{"old small", "old big", "standard", "slate", "grid", "tiles"}  type             "standard" # Defines the pavement type.
```
Comment (DavidMansolino) [inline projects/appearances/protos/Pavement.proto]: You are still using `standard` here.

File: projects/appearances/protos/SolarCell.proto [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+    }
+    roughnessMap ImageTexture {
+      url [
+        "textures/solar_cell/solar_cell_roughness.jpg"
+      ]
+    }
+    metalness 0
```
Comment (DavidMansolino) [inline projects/appearances/protos/SolarCell.proto]:
```suggestion
```

This is useless if you define a metalness map.

File: projects/appearances/protos/StonePavement.proto [hunk 23]
```diff
@@ -1,7 +1,8 @@
 #VRML_SIM R2020a utf8
 # license: Apache License 2.0
 # license url: http://www.apache.org/licenses/LICENSE-2.0
-# tags: static
+# tags: hidden, static
+# Backward compatibility with the deprecated 'StonePavement' PROTO (deprecated in Webots R2020a-rev2)
 # A stone pavement material.
 
 PROTO StonePavement [
```
[the remark this answers is not in the document]
Comment (DavidMansolino) [reply projects/appearances/protos/StonePavement.proto]: This doesn't seems to be fixed...

File: projects/appearances/protos/Pavement.proto [hunk 24]
```diff
@@ -0,0 +1,48 @@
+#VRML_SIM R2020a utf8
+# license: Apache License 2.0
+# license url: http://www.apache.org/licenses/LICENSE-2.0
+# A pavement material.
+# The type of pavement can be set using the `type` field.
+# The color can be overridden using the `colorOverride` field.
+
+PROTO Pavement [
+  field SFString{"black stone", "braun stone", "braun square stone", "slate", "grid", "tiles"}  type             "braun square stone" # Defines the pavement type.
```
Comment (DavidMansolino) [inline projects/appearances/protos/Pavement.proto]:
```suggestion
  field SFString{"black stone", "braun stone", "braun square stone", "grid", "slate", "tiles"}  type             "braun square stone" # Defines the pavement type.
```

Alphabetic order.

File: projects/samples/rendering/worlds/physically_based_rendering.wbt [hunk 25]
```diff
@@ -9,14 +9,21 @@ WorldInfo {
...
   near 0.049999999999999996
 }
 TexturedBackground {
 }
 TexturedBackgroundLight {
 }
+RectangleArena {
+  floorSize 20 20
+  floorAppearance Pavement {
+    type "regular"
```
Comment (DavidMansolino) [inline projects/samples/rendering/worlds/physically_based_rendering.wbt]: This type doesn't exist.

Comment (DavidMansolino) [reply projects/samples/rendering/worlds/physically_based_rendering.wbt]:
```suggestion
    type "grid"
```
